agrest/src/main/java/io/agrest/ResourceEntity.java

```diff
@@ -96,6 +102,42 @@ public void andQualifier(Expression qualifier) {
         return orderings;
     }
 
+    public SelectQuery<T> getSelect() {
+        return select;
+    }
+
+    public void setSelect(SelectQuery<T> select) {
+        this.select = select;
+    }
+
+
+    public List<T> getResult(AgObjectId parentId) {
+        return result.get(parentId);
+    }
+
+    /**
+     * Uses 0 bucket to retrieve result list of objects for the root entity
+     *
+     * @return
+     */
+    public List<T> getResult() {
+        return result.get(null);
+    }
+
+    public void addToResult(AgObjectId parentId, T object) {
+        result.computeIfAbsent(parentId, i -> new ArrayList()).add(object);
+    }
+
+    /**
+     * Stores plain result list without parent ID's for the root entity.
+     * uses 0 bucket to sore the result list.
+     *
+     * @param objects
+     */
+    public void addToResult(List<T> objects) {
+        result.computeIfAbsent(null, i -> new ArrayList()).addAll(objects);
```

Comment:
While I replaced this code with "setResult" for now (so this is no longer a problem), I still wanted to comment, as this will likely popup again. Why are copying the result list? This is extremely expensive. For all we know, the result may be 100K objects.